src/Interpreters/getOSKernelVersion.cpp

```diff
@@ -0,0 +1,31 @@
+#if defined(OS_LINUX)
+#include <Interpreters/getOSKernelVersion.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int SYSTEM_ERROR;
+}
+    
+String getOSKernelVersion()
+{
+    struct utsname os_kernel_info;
+    int buf = uname(&os_kernel_info);
+    if (buf < 0)
+    {
+        throw Exception(
+            "EFAULT buffer is not valid.",
+            ErrorCodes::SYSTEM_ERROR);
+    }
+    else
+    {
+        return String(os_kernel_info.sysname) + " " + String(os_kernel_info.release);
```

Comment:
But how it differs from `Poco::Environment::osName()` and `Poco::Environment::osVersion()`

https://github.com/ClickHouse/ClickHouse/blob/dab4c60596f4e6897c3d4d5fca1c6f470e2d16a2/programs/server/Server.cpp#L413-L416

Reply:
Maybe it is designed to natively support it. I just found that there are a lot of libraries in the contrib that contain this function.